### DISCUSSION TITLE
Shipping an application with a custom profile?

### DISCUSSION THREAD

**Discussion by swift502** [microsoft/terminal#14618] [Q&A] [answered]
Hey.

Is there a way to ship my terminal application with it's own profile?

Consider shipping an app like this:
```
app.exe
app_profile.json
```

How would I launch the (Python) terminal application using my custom app_profile, so that users wouldn't have to manually install the profile? The profile would change **tab title**, **icon**, and **tab color**.

Is it possible? Thanks!

**Comment by j4james** [comment, marked as answer]
I think what you're trying to achieve is documented in the [Proto extensions](https://github.com/microsoft/terminal/blob/main/doc/specs/Proto%20extensions-spec.md) specification.

Essentially you create a file with a json fragment for your custom profile, and then install that in a subdirectory under one of the `Microsoft\Windows Terminal\Fragments` paths defined in the spec.

**Reply by swift502** [reply]
That's right, thanks a lot!

**Comment by swift502** [comment]
These are related if anyone came across this discussion looking for the same thing:

https://github.com/microsoft/terminal/pull/7632
https://learn.microsoft.com/en-us/windows/terminal/json-fragment-extensions
